src/plugin/runtime/imports.ts

```diff
@@ -28,24 +30,31 @@ export async function getWasmImports(
   const imports: string[] = [];
   const importsObject: Record<string, Record<string, string>> = {};
 
+  const directory = path.dirname(asset.id);
+
   for (const moduleName of importNames) {
     const importNames = asset.imports[moduleName];
+    let importFound = false;
     const pkgImport =
       pkgJSON.imports?.[moduleName] || pkgJSON.imports?.[`#${moduleName}`];
 
     const importName = "_imports_" + genSafeVariableName(moduleName);
 
     // TODO: haandle pkgImport as object
     if (pkgImport && typeof pkgImport === "string") {
+      importFound = true;
       imports.push(genImport(pkgImport, { name: "*", as: importName }));
+    } else if (fs.existsSync(path.resolve(directory, moduleName))) {
```

Comment:
Perhaps we could use esm resolution to support node_modules and import maps and also fail if ESM import in wasm cannot be resolved? 

(we can use [exsolve](https://github.com/unjs/exsolve) for this)

Reply:
Nice, I'll switch to that. 

There's currently the path where imports don't have to be resolved (e.g. `resolved = false`). If we error out I think we'll break that. Do we want a mode where imports have to be resolved or is there some other way you are thinking we should implement that?

Reply:
We could use `exsolve.resolve(id, { try: true }`. Is there a case that non resolved ids might be fine?